### ISSUE TITLE
"Waiting for a handshake" forever.. 

### ISSUE THREAD

**Issue by vvzen** [arnaudjuracek/xy#4]
First of all: thanks for putting this up!
I was going to do this myself but certainly you've done a much better job. 😃 
I've got a problem just after installing: it keeps waiting for an handshake for **sooo long** (and it never receives it).. maybe the makeblock isn't sending anything back on the serial.
What it could be? How could it be fixed?
Namastè

PS: I used the firmware in the repo. But after uploading it the makeblock continues to crash on its X axis (it tries to keep moving even after it reaches the end of the guides).

![nodejs_plotter_xy_error](https://cloud.githubusercontent.com/assets/10340139/20122238/a5cde1a6-a616-11e6-9215-3308b9a11d1c.jpg)

**Comment by arnaudjuracek** [comment]
It seems to be a problem with the limit switches. 
My guess is that you don't have the exact same setup as mine

**Comment by arnaudjuracek** [comment]
Yep, you'll need the custom firmware in order to use this nodejs module : during the handshaking, node wait for a signal from the plotter, only sent when it reaches its final home position. So if the plotter can't reach its home, the handshaking won't resolve.

My guess is that you have a problem with your limit switches. A quick way to test them is to start the plotter from a center position, and quickly press with your finger the four limit switches. When a switch is pressed, the plotter shouldn't move in that direction. If so, maybe you have a problem in your wiring.

You could also have the X and Y steppers switched to one another.

If your problem persists, I'll tend to think that it is entirely due to the firmware, as the steppers shouldn't be able to run passed their limit switches with this custom firmware. Make sure that your limit switches are all correctly plugged, then try to inverse `WIDTH` / `HEIGHT` in the [firmware.ino](https://github.com/arnaudjuracek/xy/blob/master/firmware/firmware.ino#L29) so that you effectively switch from portrait to landscape.

As you are the second person (me being the first one) to use this code, it is possible that I went too far from the original config, and that my setup is wronger than yours. If so, I'll try to make the firmware works with an out-of-the-box plotter setup, to avoid those kind of problems. 

Also, thanks a lot for your feedback ! 👍

**Comment by vvzen** [comment]
Wow thank you so much for the detailed explanation!
I'll do all my best to help this great repo.
Now I'm going to first check if the my limit switches are setup correctly and then I'll go on debugging.
I'll let you know if the error in my setup (or in the firmware).
Which installation manual did you follow for the wiring of the cables? I don't know anymore which one I used, since I found at least 2 different ones.
Thanks, and 👍  from Rome!
